Branch-SDK/src/io/branch/referral/ShareLinkManager.java

```diff
@@ -228,6 +228,8 @@ public void onLinkCreate(String url, BranchError error) {
                     }
                 }
                 isShareInProgress_ = false;
+                // Cancel the dialog and context is released on dialog cancel
+                cancelShareLinkDialog(false);
```

Comment:
It was not there. The reason for adding it is 
1) on launching a sharing end point a new activity comes to the top of the stack and GC will try to recycle the current activity.
2) We cannot set `context =null` since the dialog termination is asynchrounos. if it is not closed when user press back, the share dialog will try to set the visible rows again and will result in crash.